src/services/TestConvertorService.ts

```diff
@@ -0,0 +1,80 @@
+import { Service } from 'typedi';
+import { TestResultSchemaTestTypesAsObject } from '../models';
+
+/**
+ * Service class for converting dynamo tests into broken down single test type tests.
+ */
+@Service()
+class TestConvertorService {
+	constructor() {}
+
+	public static isProcessModifyEventsEnabled(): boolean {
+		if (process.env.PROCESS_MODIFY_EVENTS !== 'true' && process.env.PROCESS_MODIFY_EVENTS !== 'false') {
+			throw Error('PROCESS_MODIFY_EVENTS environment variable must be true or false');
+		}
+		return process.env.PROCESS_MODIFY_EVENTS === 'true';
+	}
+
+	public static expandRecords(record: any): TestResultSchemaTestTypesAsObject[] {
+		console.log('expanding records');
+		const splitRecords: TestResultSchemaTestTypesAsObject[] = [];
+		const templateRecord = Object.assign({}, record);
+		Object.assign(templateRecord, {});
+		if (Array.isArray(record.testTypes)) {
+			record.testTypes?.forEach((testType: any, i: number, array: any[]) => {
+				const clonedRecord: any = Object.assign({}, templateRecord); // Create record from template
+				Object.assign(clonedRecord, { testTypes: testType }); // Assign it the test type
+				Object.assign(clonedRecord, {
+					// Assign certificate order number
+					order: {
+						current: i + 1,
+						total: array.length,
+					},
+				});
+				splitRecords.push(clonedRecord);
+			});
+		}
+
+		const flatSplitRecords: TestResultSchemaTestTypesAsObject[] = splitRecords.reduce(
+			(acc: any[], val: any) => acc.concat(val),
+			[]
+		);
+		const filteredRecords = TestConvertorService.filterCertificateGenerationRecords(flatSplitRecords);
+
+		return filteredRecords;
+	}
```

Comment:
I think this might be able to simplified. It looks like we can use a `.map here to define a new object for each of the test types. I think we can the remove all cloning done by object.assign. 

Maybe something like?

```
public static expandRecords(record: any): TestResultSchemaTestTypesAsObject[] {
 if (!Array.isArray(record.testTypes)) {
   return [];
 }

 const expandedRecords = record.testTypes.map((testType: any, i: number) => ({
   ...record,
   testTypes: testType,
   order: {
     current: i + 1, 
     total: record.testTypes.length
   }
 }));

 return TestConvertorService.filterCertificateGenerationRecords(expandedRecords);
}
```